hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieStreamingSink.scala

```diff
@@ -78,26 +82,61 @@ class HoodieStreamingSink(sqlContext: SQLContext,
       log.error("Async clustering service shutdown unexpectedly")
       throw new IllegalStateException("Async clustering service shutdown unexpectedly")
     }
+
+    // get the latest checkpoint from the commit metadata to check if the microbatch has already been prcessed or not
+    if (metaClient != null) {
```

Comment:
Could we extra the checking logic to a function? e.g., checkCanSkipBatch(xxx)

Reply:
In restart, the metaClient is null (because it seems only be initialized after the first batch). Then we will not check lastBatchId for the first batch after start (or restart after crash). 

And seems the whole checking logic is only necessary for the first batch, right? If so, could we save the checking costs by only doing it once for the first batch?

Reply:
That's a very good point! The intention is to check it in every batch. Actually, i did initialize the metaclient in my base commit but i missed it after rebase. Thanks for catching it. I am now initializing the metaclient once again just reloading active timeline in each batch.